### PR TITLE
Actually fixes temperature pump exploit. Removes limit as it's redundant now.

### PR DESCRIPTION
**Pull request by Pickle-Coding**
## About The Pull Request
Temperature pumps were bugged and generated heat while exchanging heat. This fixes that and removes the limits as they were only used to stop people from creating infinite heat.

## Why It's Good For The Game
Exploit bad, and the limit was below a temperature that can be obtained legitimately, which was annoying.

## Changelog

:cl: AccountName5
balance: Temperature pump power consumption no longer increases exponentially. Removes maximum operational temperature as it is now obsolete.
fix: Fixed temperature pump exploit. It should no longer treat output as having a much lower heat capacity than it actually has.
/:cl:

### CODE REVIEW

**Comment by 170141183460469231731687303715884105727** [comment]
I'm sorry about all the useless commits from my previous PRs. I accidentally created the branch from another branch I was using.

**Comment by 170141183460469231731687303715884105727** [comment]
There's also another bug where temperature pump doesn't exchange heat if temperature delta is below 1K. I don't know what the cause is for that though.

**Comment by Ghilker** [comment]
The fix is good but i don't know about the other stuff since you can theoretically go over the limit anyway and the power usage you can lower it by just setting it to check on the output temperature instead of the input, but i don't like it butchered like that, tho i'm working on making baseline procs for atmos that will include power consumption so i guess it's fine?

**Comment by 170141183460469231731687303715884105727** [comment]
I don't think it would be problematic to remove the limit since the exploit gets fixed. You'd probably need like hundreds, if not thousands of t3 canisters to reach a temperature that gets problematic, and millions of moles of plasma and o2. You'd need multiple burnt burn mixes to heat a single burn mix in the next generation to the desired temperature to be profitable.

I could make power consumption linear though, so there would be a reason to use heat exchangers over temperature pumps.

**Comment by 170141183460469231731687303715884105727** [comment]
I could make it some constant + coolant_heat_amount * some_very_small_number.

**Comment by 170141183460469231731687303715884105727** [comment]
I couldn't convince the potato about removing temperature pump limit, so sadly I'll add it back in tomorrow.

File: code/modules/atmospherics/machinery/components/binary_devices/temperature_pump.dm [hunk 1]
```diff
@@ -44,16 +44,16 @@
 
 	var/coolant_temperature_delta = remove_input.temperature - remove_output.temperature
 
-	if(coolant_temperature_delta > 0 && remove_input.temperature < 1e8)
+	if(coolant_temperature_delta > 0)
 		var/input_capacity = remove_input.heat_capacity()
-		var/output_capacity = air_output.heat_capacity()
+		var/output_capacity = remove_output.heat_capacity()
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/components/binary_devices/temperature_pump.dm:49]: This is depressing

**Comment by LemonInTheDark** [comment]
> 
> 
> I couldn't convince the potato about removing temperature pump limit, so sadly I'll add it back in tomorrow.

I was under the impression that the heat pump actually pumped heat. apparently it doesn't. currently considering if it even needs to exist.
image

**Comment by 170141183460469231731687303715884105727** [comment]
I might rework temperature pumps in a future PR to make it more different than heat exchangers, and maybe turn the heat exchangers into a single binary device that has the control of current temperature pumps.

**Pull request merged**